### DISCUSSION TITLE
Will this setup work?

### DISCUSSION THREAD

**Discussion by Tronnic** [Ysurac/openmptcprouter#3063] [Q&A]
Hello everyone, 
I have access to 3x 100mbps internet connections. Will this setup work? 

![image](https://github.com/Ysurac/openmptcprouter/assets/2569173/9e3cdde2-68a5-450b-804a-b9363449b947)

We're trying to share our internet-connections (all family in the same house) and bundle them to have a shared 300 mbps connection. 

Is this a possible setup?

Thanks for any help! I'm fairly new to networking so I'm unsure.

**Comment by iambenmitchell** [comment]
You only need 1 switch and 4 routers. 

1 router on each of the 100mbps connections + openmptcp router.

**Reply by iambenmitchell** [reply]
Will your openmptcp router have 4 Ethernet ports? Or will you be using VLAN switch? 

(I would avoid using USB Ethernet adapters)

**Reply by iambenmitchell** [reply]
![image](https://github.com/Ysurac/openmptcprouter/assets/27426792/2ecf8117-af05-419d-809c-f00b9917a19d)

**Reply by Tronnic** [reply]
Thank you for your reply! We only have 1 LAN-Cable from each household going to the central switch. So the leftmost switches are the switches located in the households. The right switch is the "central" switch where the openmptcprouter will be plugged in. I do have USB 3 Ethernet Adapters but was thinking about using VLAN to make it simpler (if that works well or even better). 

So I'm assuming my general thinking was correct in the graphic I posted, yes? :)

**Reply by Ysurac** [reply]
No need for VLAN or multiple ethernet adapters if each routers are in different subnet, if you can reach/ping all routers from a PC, you can install virtualbox for example and test in a VM. Client will have to use VM IP as gateway.
If your OpenMPTCProuter have only one ethernet card, then "virtual" ethernet interfaces are created using MacVLAN.

**Reply by iambenmitchell** [reply]
If you have a smart switch that can do VLANs, I would recommend doing it that way. 

Your thinking is correct, but don’t understand your internet router and WLAN router diagrams?  

I would connect your ISP routers directly to the openmptcp router and disable any WiFi on the ISP routers. That way the connection is dedicated to openmptcp. 

You’ll then need to have WiFi access to your OpenMPTCP LAN, these access points will need to be on the 192.168.100.x LAN.. I think that’s what you’re saying in your diagram but I’m not sure

**Reply by iambenmitchell** [reply]
When I say connect directly, I mean either VLAN or a dedicated Ethernet port if the machine you’re using to run Openmptcp has 4 Ethernet ports. 

If it doesn’t have 4 ports then use VLAN. There’s other ways of doing it such as using gateway IPs and MACLANs or USB Ethernet dongles.. but i would avoid these and either use a 4 port PCIe NIC or VLANs

**Reply by Tronnic** [reply]
> Your thinking is correct, but don’t understand your internet router and WLAN router diagrams?

To explain my diagram:

Each household has a Internet Router coming out of the telephone line that will be set to a static IP, in the same household is also a switch, and another Wifi Router so every household has its own network and own wifi where all the households devices will be connected. Then, there is a LAN cable plugged into the switch that goes down to the basement where there is a "central switch".

**Comment by Tronnic** [comment]
> No need for VLAN or multiple ethernet adapters if each routers are in different subnet, if you can reach/ping all routers from a PC, you can install virtualbox for example and test in a VM. Client will have to use VM IP as gateway. If your OpenMPTCProuter have only one ethernet card, then "virtual" ethernet interfaces are created using MacVLAN.

@Ysurac I'm sorry if this is a total noob question, but with Subnet, do you mean the subnet-mask or is it fine to leave them all on 255.255.255.0 but have the IP-Ranges and IP-Gateways completley seperated as I've drawn in the diagram. 

I've updated my diagram to reflect the exact way I'm trying to connect everything:

![image](https://github.com/Ysurac/openmptcprouter/assets/2569173/055b9e72-3869-43f5-b950-4a3acdb29f78)
Each line represents a single LAN-Cable

**Reply by Ysurac** [reply]
In your diagram, 255.255.255.0 (/24) is ok, 192.168.20.x, 192.168.10.x,... are different subnets. In OpenMPTCProuter you will have to set interfaces (or virtual interfaces) in each subnet and use router IP as gateway for each.
You know on which arch you want to host OpenMPTCProuter ? (Raspberry PI, Virtualbox, VMware, dedicated x86_64,...)

**Reply by Tronnic** [reply]
@Ysurac  I am going to use a Raspberry Pi 4b+ with 4GB of RAM (still have one laying around). But if there are better solutions, I would be up for it :) Also thank you for your answer! I'm learning so much new things about networking since I started to think about this project. Great work so far, lots of thanks to you for your efforts! <3

Also, how resource-intensive is the vps side? I rented a 2 core (shared vCPU x86) vserver from hetzner. They also have a cheaper 1 core option. What is recommended? :)

**Reply by Ysurac** [reply]
You can reach about 550Mb/s with encryption enabled on a RPI4 (maybe more on latest OpenMPTCProuter snapshot, not yet tested on RPI4). A RPI5 can reach about 900Mb/s with encryption (limited by ethernet interface at 1Gb/s).
Most VPS have AES hardware enabled, so it's less CPU intensive. In most case 1vCPU should be enough if the server host is quite fast, difficult to say...

**Reply by iambenmitchell** [reply]
Pi4 works well, that’s what I used to use. I’d recommend using a managed switch (VLAN switch) and setting it up that way. Gigabit ones cost around £20-£30. 

Alternatively you can do what @Ysurac said and do the macvlan with gateways.

**Comment by Tronnic** [comment]
Got it working! "Almost" Everything seems to be running smoothly. Currently only running with 1 internet-connection. The next is coming online on friday, and the third one will probably be set up on saturday. The only issue I'm currently facing is that I cannot watch Disney+ anymore. I've read some posts in here trying to add disney+to OMR Bypass and also adding some domains, but nothing seems to be working for me (Germany). Any way to fix this?

**Reply by iambenmitchell** [reply]
Issues with streaming sites are caused when you use a data centre IP (in this case your VPS) 

If you have a friend who has a fast internet connection, I’d recommend asking if they could host you a VPS in their house otherwise you’ll continue to run into issues like this and be constantly solving captchas. 

For now you can just setup the OMR bypass so it sends Disney traffic down one of your 100mbps connections. I can’t remember how to do this but you’ve done it already? I’m not sure why it wouldn’t work. Make sure you select the correct interface

**Reply by Tronnic** [reply]
yeah I'm trying since a few hours now, I've added a LOT of ASN's and domains, but nothing seems to be working. Also, running into issues logging into Leauge of Legends. Seems like Hetzner is blocked by many, I hope I can find a VPS that isn't blocked. :( I don't have a friend with a spare gigabit connection unfortunately.

**Reply by Ysurac** [reply]
You added Disneyplus using Protocol/services section in OMR-ByPass ?
You are using OpenMPTCProuter IP as DNS on client ? You may have to reset DNS cache (or reboot) client.
